src/core/thread/mesh_forwarder.cpp

```diff
@@ -909,6 +911,19 @@ void MeshForwarder::HandleSentFrame(Mac::TxFrame &aFrame, otError aError)
         mMessageNextOffset = mSendMessage->GetLength();
 #endif
     }
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
```

Comment:
If we do this check in `MeshForwarder` we are restricting the check to only direct tx requests. 
If we do the same check in `Mac`, we can cover acks to any type of tx (in particular I am considering `kOperationTransmitOutOfBandFrame`).

Another suggestion: Maybe we can re-use the `DataPollSender::CheckFramePending(RxFrame)` method?

Thoughts?

Reply:
Moved into MAC layer and changed to reuse CheckFramePending(), but leave out of band frame handling for future because we are considering remove it.